src/base.jl

```diff
@@ -1,14 +1,11 @@
 ### Struct and different Power System constructors depending on the data provided ####
 
-struct PowerSystem{T <: Union{Nothing,Array{ <: ThermalGen,1}},
-                   R <: Union{Nothing,Array{ <: RenewableGen,1}},
-                   H <: Union{Nothing,Array{ <: HydroGen,1}},
-                   L <: ElectricLoad,
+struct PowerSystem{L <: ElectricLoad,
                    B <: Union{Nothing,Array{ <: Branch,1}},
                    S <: Union{Nothing,Array{ <: Storage,1}}
                    }
     buses::Array{Bus,1}
-    generators::NamedTuple{(:thermal, :renewable, :hydro), Tuple{T, R, H}}
+    generators::GenClasses
```

Comment:
I like the change but I'm not sure what this will break in PowerSimulations.

Reply:
This change will break PowerSimulations. We had an approach like this before but it is preferable to have a NamedTuple so it is possible to make iterations.